SilverStripe 4 upgrade

jsirish opened the pull request.

jsirish commented:
@muskie9 this is in pretty good shape. integrated a number of things from your PR as well. I think we should be able to merge and create issues as we run across them.

I did however have the image upload error on a fresh install from local composer after a 5.6 upgrade. Might be worth a test against a fresh install of alpha7

muskie9 commented on src/FlexSlider.php:
```diff
@@ -59,11 +88,7 @@ public function updateCMSFields(FieldList $fields)
-                $config->addComponent(new GridFieldBulkUpload());
-                $config->addComponent(new GridFieldBulkManager());
-            }
+            //$config->addComponent(new GridFieldOrderableRows('SortOrder'));
```
With gridfieldextensions having a 4 compatible branch now, does orderable work?

jsirish replied:
did a composer update, was an update for extensions - works now!

muskie9 commented:
Should we update calls such as `$before = (method_exists($this->owner->ClassName, 'flexSliderBeforeAction'))` to something like `$before = $this->owner->hasMethod('flexSliderBeforeAction')`? I'm not sure if there's a performance benefit to either.

jsirish commented:
I've updated the JS call, and switched in `hasMethod` for `method_exists`. 

Verified it's working on the front end, though you need to add the `<% include FlexSlider %>` to Page.ss in Simple theme, and remove the two JS calls in the top level Page.ss. Thumbnail nav works as well.

All tests pass locally and on Travis other than SQLite.

muskie9 commented:
This checked out locally, I've restarted tests to see if the sqlite will pass. The only thing I ran into is the Simple theme jquery inclusion caused the frontend to fail. Removing that inclusion from the main `Page.ss` file fixed it.
